pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1014,12 +959,7 @@ public void writeBrokerDataOnZooKeeper(boolean force) {
             if (needBrokerDataUpdate() || force) {
                 localData.setLastUpdate(System.currentTimeMillis());
 
-                try {
-                    zkClient.setData(brokerZnodePath, localData.getJsonBytes(), -1);
-                } catch (KeeperException.NoNodeException e) {
-                    ZkUtils.createFullPathOptimistic(zkClient, brokerZnodePath, localData.getJsonBytes(),
-                            ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL);
-                }
+                brokerDataLock.updateValue(localData).join();
```

Comment:
same here, we should use time bounded get instead join . this can cause deadlock in system if future never completes.